roles/kubernetes/preinstall/tasks/0090-etchosts.yml

```diff
@@ -1,32 +1,30 @@
 ---
-- name: Hosts | update inventory in hosts file
-  when: populate_inventory_to_hosts_file
-  block:
-    - name: Hosts | create list from inventory
-      set_fact:
-        etc_hosts_inventory_block: |-
-          {% for item in (groups['k8s_cluster'] + groups['etcd'] | default([]) + groups['calico_rr'] | default([])) | unique -%}
-          {% if 'access_ip' in hostvars[item] or 'ip' in hostvars[item] or 'ansible_default_ipv4' in hostvars[item] -%}
-          {{ hostvars[item]['access_ip'] | default(hostvars[item]['ip'] | default(hostvars[item]['ansible_default_ipv4']['address'])) }}
-          {%- if ('ansible_hostname' in hostvars[item] and item != hostvars[item]['ansible_hostname']) %} {{ hostvars[item]['ansible_hostname'] }}.{{ dns_domain }} {{ hostvars[item]['ansible_hostname'] }} {% else %} {{ item }}.{{ dns_domain }} {{ item }} {% endif %}
+- name: Hosts | create hosts list from inventory
+  set_fact:
+    etc_hosts_inventory_block: |-
+      {% for item in (groups['k8s_cluster'] + groups['etcd'] | default([]) + groups['calico_rr'] | default([])) | unique -%}
+      {% if 'access_ip' in hostvars[item] or 'ip' in hostvars[item] or 'ansible_default_ipv4' in hostvars[item] -%}
+      {{ hostvars[item]['access_ip'] | default(hostvars[item]['ip'] | default(hostvars[item]['ansible_default_ipv4']['address'])) }}
+      {%- if ('ansible_hostname' in hostvars[item] and item != hostvars[item]['ansible_hostname']) %} {{ hostvars[item]['ansible_hostname'] }}.{{ dns_domain }} {{ hostvars[item]['ansible_hostname'] }} {% else %} {{ item }}.{{ dns_domain }} {{ item }} {% endif %}
 
-          {% endif %}
-          {% endfor %}
-      delegate_to: localhost
-      connection: local
-      delegate_facts: yes
-      run_once: yes
+      {% endif %}
+      {% endfor %}
+  delegate_to: localhost
+  connection: local
+  delegate_facts: yes
+  run_once: yes
+  when: populate_inventory_to_hosts_file
 
-    - name: Hosts | populate inventory into hosts file
-      blockinfile:
-        path: /etc/hosts
-        block: "{{ hostvars.localhost.etc_hosts_inventory_block }}"
-        state: present
-        create: yes
-        backup: yes
-        unsafe_writes: yes
-        marker: "# Ansible inventory hosts {mark}"
-        mode: 0644
+- name: Hosts | populate inventory into hosts file
+  blockinfile:
+    path: /etc/hosts
+    block: "{{ hostvars.localhost.etc_hosts_inventory_block }}"
+    state: "{{ 'present' if populate_inventory_to_hosts_file else 'absent' }}"
+    create: yes
+    backup: yes
+    unsafe_writes: yes
+    marker: "# Ansible inventory hosts {mark}"
+    mode: 0644
```

Comment:
HI @rptaylor 

When set the `populate_inventory_to_hosts_file` to false, the block will fail with logs:

```
fatal: [kay171]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'etc_hosts_inventory_block'. 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'etc_hosts_inventory_block'\n\nThe error appears to be in '/root/oss/kubespray/roles/kubernetes/preinstall/tasks/0090-etchosts.yml': line 18, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Hosts | populate inventory into hosts file\n  ^ here\n"}
fatal: [kay172]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'etc_hosts_inventory_block'. 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'etc_hosts_inventory_block'\n\nThe error appears to be in '/root/oss/kubespray/roles/kubernetes/preinstall/tasks/0090-etchosts.yml': line 18, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Hosts | populate inventory into hosts file\n  ^ here\
```
The `etc_hosts_inventory_block`  is not defined.

Thanks